src/components/xy_chart/xy_chart.js

```diff
@@ -0,0 +1,202 @@
+        component.onParentMouseLeave(event);
+      }
+    });
+  }
+}
+
+class XYChart extends PureComponent {
+  state = {
+    mouseOver: false,
+  };
+  colorIterator = 0;
+  _xyPlotRef = React.createRef();
+
+  _renderChildren = (child, i) => {
+    const { prototype } = child.type;
+    // Avoid applying chart props to non series children
+    if (!(prototype instanceof AbstractSeries)) {
+      return child;
+    }
+
+    const props = {
+      id: `chart-${i}`,
+    };
+
+    if (!child.props.color) {
+      props.color = VISUALIZATION_COLORS[this.colorIterator];
+
+      this.colorIterator++;
+      if (this.colorIterator > VISUALIZATION_COLORS.length - 1) this.colorIterator = 0;
+    }
+
+    return React.cloneElement(child, props);
+  };
+  // canShowCrosshair = () => {
+  //   const { crosshairValue, showCrosshair } = this.props;
+  //   const { mouseOver } = this.state;
+  //   return showCrosshair && ( mouseOver || crosshairValue !== undefined)
+  // }
+
+  render() {
+    const {
+      width,
+      height,
+      xType,
+      yType,
+      stackBy,
+      errorText,
+      children,
+      xDomain,
+      yDomain,
+      yPadding,
+      xPadding,
+      animation,
+      showDefaultAxis,
+      showCrosshair,
+      showBrush,
+      brushOrientation,
+      onBrushEnd,
+      orientation,
+      crosshairValue,
+      onCrosshairUpdate,
+      ...rest
+    } = this.props;
+
+    if (!children || errorText) {
+      return <StatusText text={errorText} width={width} height={height} />;
+    }
+
+    this.colorIterator = 0;
+    const Crosshair =
+      orientation === EuiXYChartUtils.ORIENTATION.HORIZONTAL ? EuiCrosshairY : EuiCrosshairX;
+    return (
+      <div {...rest}>
+        <XYExtendedPlot
+          ref={this._xyPlotRef}
+          dontCheckIfEmpty
+          onMouseMove={this._onMouseMove}
+          onMouseLeave={this._onMouseLeave}
+          width={width}
+          animation={animation}
+          height={height}
+          margin={2}
+          xType={xType}
+          yType={yType}
+          xDomain={xDomain}
+          yDomain={yDomain}
+          stackBy={stackBy}
+          yPadding={yPadding}
+          xPadding={xPadding}
+        >
+          {React.Children.map(children, this._renderChildren)}
+          {showDefaultAxis && <EuiDefaultAxis orientation={orientation} />}
+          {showCrosshair && (
+            <Crosshair crosshairValue={crosshairValue} onCrosshairUpdate={onCrosshairUpdate} />
+          )}
+
+          {showBrush && <EuiBrush onBrushEnd={onBrushEnd} orientation={brushOrientation}/>}
+        </XYExtendedPlot>
+      </div>
+    );
+  }
+}
+XYChart.displayName = 'EuiXYChart';
+
+XYChart.propTypes = {
+  /** The initial width of the chart. */
+  width: PropTypes.number.isRequired,
+  /** The initial height of the chart. */
+  height: PropTypes.number.isRequired,
+  /** The orientation of the chart. */
+  orientation: PropTypes.oneOf([ HORIZONTAL, VERTICAL ]),
```

Comment:
In talking with Tim, we decided this was most likely an unused feature of the current charts, very broken currently and should most likely not be in the new charts. Did this change?

Reply:
Yeah, I think that's what I said that time :D But since we already got this implemented, it seems Marco also got the actual issues solved (which were mainly around the way we do selection and show the crosshair in the "wrong direction"). So if it's implemented and useable I am all for not needing to break anything there :)

Reply:
If those points hold true I’m fine with that then. There seems to be some issue with the crosshairs implementation though so this will hinge then on where that falls